docs/COMPLEX_COMPOSITION.md

```diff
@@ -18,6 +18,65 @@ Where Guard rule is an umbrella term for clause, query block, `when` block or na
 
 1. Named rule blocks allow for re-usability, improved composition, and reduced verbosity and repetition. This document will focus on demonstrating these features of named rule blocks in-depth.
 
+## Clause evaluation
+
+Clauses are evaluated on the entire query result. Let us look at some examples to understand this better.
+
+```
+let s3_buckets = Resources.*[ Type == "AWS::S3::Bucket" ]
+
+rule check_s3_bucket when %s3_buckets !empty {
+  %s3_buckets.Properties.AccessControl exists               << Clause 1 >>
+  %s3_buckets.Properties.LoggingConfiguration exists        << Clause 2 >>
+}
+```
+
+In the above example `check_s3_bucket` block evaluates to the following,
+- When the input template has S3 buckets,
+  - All S3 buckets should have `AccessControl` property defined **AND** All S3 buckets should have `LoggingConfiguration` defined.
+
+Let us look at the next example where `Clause 1` and `Clause 2` are joined using disjunction to solidify your understanding,
+
+<pre>
```

Comment:
copy paste error?

Reply:
That is a valid formatting tag, https://gist.github.com/mufid/4062574.